packages/jest-worker/src/Farm.ts

```diff
@@ -44,41 +46,64 @@ export default class Farm {
     }
   }
 
-  doWork(method: string, ...args: Array<any>): Promise<unknown> {
-    return new Promise((resolve, reject) => {
-      const computeWorkerKey = this._computeWorkerKey;
-      const request: ChildMessage = [CHILD_MESSAGE_CALL, false, method, args];
+  doWork(
+    method: string,
+    ...args: Array<any>
+  ): PromiseWithCustomMessage<unknown> {
+    const customMessageListeners = new Set<OnCustomMessage>();
 
-      let worker: WorkerInterface | null = null;
-      let hash: string | null = null;
+    const addCustomMessageListener = (listener: OnCustomMessage) => {
+      customMessageListeners.add(listener);
+      return () => {
+        customMessageListeners.delete(listener);
+      };
+    };
+
+    const onCustomMessage: OnCustomMessage = message => {
+      customMessageListeners.forEach(listener => listener(message));
+    };
 
-      if (computeWorkerKey) {
-        hash = computeWorkerKey.call(this, method, ...args);
-        worker = hash == null ? null : this._cacheKeys[hash];
-      }
+    const promise: PromiseWithCustomMessage<unknown> = new Promise(
+      (resolve, reject) => {
+        const computeWorkerKey = this._computeWorkerKey;
+        const request: ChildMessage = [CHILD_MESSAGE_CALL, false, method, args];
 
-      const onStart: OnStart = (worker: WorkerInterface) => {
-        if (hash != null) {
-          this._cacheKeys[hash] = worker;
+        let worker: WorkerInterface | null = null;
+        let hash: string | null = null;
+
+        if (computeWorkerKey) {
+          hash = computeWorkerKey.call(this, method, ...args);
+          worker = hash == null ? null : this._cacheKeys[hash];
         }
-      };
 
-      const onEnd: OnEnd = (error: Error | null, result: unknown) => {
-        if (error) {
-          reject(error);
+        const onStart: OnStart = (worker: WorkerInterface) => {
+          if (hash != null) {
+            this._cacheKeys[hash] = worker;
+          }
+        };
+
+        const onEnd: OnEnd = (error: Error | null, result: unknown) => {
+          customMessageListeners.clear();
+          if (error) {
+            return reject(error);
```

Comment:
`return` does nothing

Reply:
Just keeping `reject(error)` was throwing linting errors, so had to add a return behind it